packages/dd-trace/src/encode/0.4.js

```diff
@@ -263,6 +271,45 @@ class AgentEncoder {
     }
   }
 
+  _encodeObject (bytes, value, circularReferencesDetector = new Set()) {
+    circularReferencesDetector.add(value)
+    if (Array.isArray(value)) {
+      return this._encodeObjectAsArray(bytes, value, circularReferencesDetector)
+    } else if (typeof value === 'object') {
+      return this._encodeObjectAsMap(bytes, value, circularReferencesDetector)
+    } else if (typeof value === 'string' || typeof value === 'number') {
+      this._encodeValue(bytes, value)
+    }
+  }
+
+  _encodeObjectAsMap (bytes, value, circularReferencesDetector) {
+    const keys = Object.keys(value)
+    const validKeys = keys.filter(
+      key =>
+        typeof value[key] === 'string' || typeof value[key] === 'number' ||
+        (typeof value[key] === 'object' && !circularReferencesDetector.has(value[key])))
+
+    this._encodeMapPrefix(bytes, validKeys.length)
```

Comment:
should you check if there are valid keys before encode?
```suggestion
    if (validKeys.length > 0) {
      this._encodeMapPrefix(bytes, validKeys.length)
    }
```

Reply:
I think that empty objects and arrays could be sent, it is tested here:

https://github.com/DataDog/dd-trace-js/pull/4287/files#diff-56980413b3ad2238eef2cef72c41f90b63dcc811417d27f37e3f62b4040b9fc0R280-R283